src/lib/analytics/AnalyticsClass.js

```diff
@@ -1,30 +1,59 @@
 // @flow
-import { assign, debounce, forIn, get, isEmpty, isNumber, isString, pick, remove, toLower, values } from 'lodash'
+import {
+  assign,
+  debounce,
+  forIn,
+  get,
+  isEmpty,
+  isNumber,
+  isString,
+  isUndefined,
+  negate,
+  once,
+  pick,
+  pickBy,
+  remove,
+  toLower,
+  values,
+} from 'lodash'
+
 import { isMobileReactNative } from '../utils/platform'
 import { LogEvent } from '../logger/pino-logger'
 import { ExceptionCategory } from '../logger/exceptions'
 import { ANALYTICS_EVENT, ERROR_LOG } from './constants'
 
-const savedErrorMessages = new WeakMap()
-
 export class AnalyticsClass {
   apis = {}
 
-  constructor(apis, rootApi, Config, loggerApi) {
+  savedErrorMessages = new WeakMap()
+
+  constructor(apisFactory, rootApi, Config, loggerApi) {
     const { sentryDSN, amplitudeKey, env } = Config
-    const { amplitude, sentry, fullStory } = apis
     const logger = loggerApi.child({ from: 'analytics' })
     const options = pick(Config, 'sentryDSN', 'amplitudeKey', 'version', 'env', 'phase')
 
-    assign(this.apis, apis)
-    assign(this, options, { logger, rootApi, loggerApi })
+    const initApis = () => {
+      const apis = apisFactory()
+      const { amplitude, sentry, fullStory } = apis
+
+      assign(this.apis, apis)
+
+      this.isSentryEnabled = sentry && sentryDSN
+      this.isAmplitudeEnabled = amplitude && amplitudeKey
+      this.isFullStoryEnabled = fullStory && env === 'production'
+    }
+
+    this.initAnalytics = once(async () => {
+      const { initAnalytics } = this.constructor.prototype
```

Comment:
too much , lets  do it without this voodoo
just call initAnalytics. we dont need to enforce once etc.
keep it simple

Reply:
done